### PR TITLE
#164523189 User can respond to meetup rsvp

### PR DESCRIPTION
**Pull request by Bakley**
## What does this PR do?
- add rsvp models with relation to meetup and user
- registered the rsvp to admin to display on admin panel
- fix the social auth

## What are the tasks to be completed?
- Ensure the following endpoints are working:-
```
1. POST api/<meetup-id>/rsvp
2. GET /api/response
```

### Local Development Setup
- First Create python virtual env
```
 $ virtualenv -p python .venv
```
- Install Requirements
```
 $ pip install -r requiremts.txt
```
- Copy .env-example to .env and set config
```
 $ cp .env-example .env
```
- Create postgres database
```
 $ sudo su postgres
 $ psql
 postgres=# CREATE USER django WITH PASSWORD 'password';
 postgres=# ALTER ROLE django SET client_encoding TO 'utf8';
 postgres=# ALTER ROLE django SET default_transaction_isolation TO 'read committed';
 postgres=# ALTER ROLE django SET timezone TO 'UTC';
 postgres=# CREATE DATABASE drf;
 postgres=# GRANT ALL PRIVILEGES ON DATABASE drf TO django;
 postgres=# \q
 $ exit
```
- Create admin user
    1. cd api/
    2. python manage.py createsuperuser
- Create User using your email and password

- Run Server
```
$ python manage.py runserver
```
a. After cloning the repo:-
```
git checkout ft-users-rsvp-meetups-164523189
```
b. Create meetups

- Follow the instructions [here](https://github.com/Ogutu-Brian/questioners/pull/6) to create meetups

c. Create a RSVP
```
{
	"response": "Yes"
}
```
- Use postman as shown below to test
get_all_questioner_meetup_rsvp_response
rsvp_docu

### CODE REVIEW

File: api/auth/views.py [hunk 1]
```diff
@@ -86,11 +86,17 @@ class SocialSigUpView(APIView):
     """
     
     permission_classes = [permissions.AllowAny]
+    serializer_class = serializers.SocialAuthSerializer
 
     def post(self, request):
```
Comment (Bakley) [inline api/auth/views.py:91]: @matthenge Check here what changed to allow social signup

**Review by IssaIan — CHANGES_REQUESTED** [review]
@bakley change the swagger link to point at heroku staging url

File: api/meetups/serializers.py [hunk 2]
```diff
@@ -18,3 +28,27 @@ class TagSerializer(serializers.ModelSerializer):
...
+
+    def validate_response(self, value):
+        """
+        Validate User response
+        """
+
+        if "yes" not in value.lower():
+            if "no" not in value.lower():
+                if "maybe" not in value.lower():
+                    print(value)
```
Comment (a-braham) [inline api/meetups/serializers.py]: You forgot to remove the print statement

**Comment by Bakley** [comment]
> @Koin change the swagger link to point at heroku staging url

Yes changing asap

File: api/meetups/tests/test_meetups.py [hunk 3]
```diff
@@ -307,3 +318,95 @@ def test_invalid_image_url(self) -> None:
...
+            url,
+            data=json.dumps({
+                "response": "Yes"
+            }),
+            content_type="application/json"
+        )
+
+        return response
+
+    def post_rsvp_without_response(self):
```
Comment (a-braham) [inline api/meetups/tests/test_meetups.py]: The test method should start with the key word test like `test_post_rsvp_without_response(self)`

File: api/meetups/tests/test_meetups.py [hunk 4]
```diff
@@ -307,3 +318,95 @@ def test_invalid_image_url(self) -> None:
...
+        self.meetup = Meetup.objects.create(
+            title='Test Driven Development',
+            body='Developers need to discusss the correct approach of doing test driven development',
+            location='Andela Campus',
+            creator=self.admin,
+            scheduled_date=timezone.now() + timezone.timedelta(days=3)
+        )
+        self.meetup.save()
+
+    def login_user(self, email="admin@questioner.com", password="@Escobar123"):
```
Comment (a-braham) [inline api/meetups/tests/test_meetups.py:334]: I don't see where this method is being used.

**Review by matthenge — COMMENTED** [review]
@Bakley  Have you excluded swagger.py from tests?

**Review by a-braham — COMMENTED** [review]
That's a good job

File: api/meetups/serializers.py (hunk 2 above)
Comment (Bakley) [reply api/meetups/serializers.py]: Ahh thanks @a-braham , let me get rid of it

File: api/auth/views.py [hunk 5]
```diff
@@ -86,11 +86,17 @@ class SocialSigUpView(APIView):
...
-        auth = get_authorization_header(request).split()
+        """
+        User sign up with Google
+        """
         not_active = True
-        token = auth[1]
+        serializer = self.serializer_class(data=request.data)
+        serializer.is_valid(raise_exception=True)
+        token = serializer.data.get("id_token")
+        print(token )
```
Comment (matthenge) [inline api/auth/views.py]: You forgot a print statement in the code

**Review by matthenge — DISMISSED** [review]

File: api/auth/views.py (hunk 5 above)
Comment (Bakley) [reply api/auth/views.py]: Ooh, thanks for catching that

**Review by Ogutu-Brian — APPROVED** [review]
Looks Good to me

**Review by IssaIan — COMMENTED** [review]
@Ogutu-Brian had requested us to remove the json editor setting from the swagger settings in the settings file. Otherwise good work.

**Comment by Bakley** [comment]
> @Ogutu-Brian had requested us to remove the json editor setting from the swagger settings in the settings file. Otherwise good work.

@IssaIan Done

**Review by IssaIan — APPROVED** [review]

**Pull request merged**